File: string__reverse_words.js
function reverseWords(str) {
    let buff = [],
        l = str.length,
        tokenReadPos = l - 1;

    while (tokenReadPos >= 0) {
        if (str[tokenReadPos] === ' ') {
            buff.push(str[tokenReadPos--]);
        } else {
            let wordEnd = tokenReadPos;
            while (tokenReadPos >= 0 && str[tokenReadPos] !== ' ')
                tokenReadPos--;
            let wordReadPos = tokenReadPos + 1;
            while (wordReadPos <= wordEnd)
                buff.push(str[wordReadPos++]);
        }
    }

    return buff.join('');
}

console.log(reverseWords('Do or do not, there is no try.'));

File: string__reverse_words.py
def reverseWords(str):
    length = len(str)
    tokenReadPos = length - 1
    buff = []
    while tokenReadPos >= 0:
        if str[tokenReadPos] == ' ':
            buff.append(str[tokenReadPos])
            tokenReadPos -= 1
        else:
            wordEnd = tokenReadPos
            while tokenReadPos >= 0 and str[tokenReadPos] != ' ':
                tokenReadPos -= 1
            wordReadPos = tokenReadPos + 1
            while wordReadPos <= wordEnd:
                buff.append(str[wordReadPos])
                wordReadPos += 1
    return ''.join(buff)


str = 'Do or do not, there is no try.'
print(reverseWords(str))